src/logic/safe/store/middleware/notificationsMiddleware.ts

```diff
@@ -68,97 +69,97 @@ const sendAwaitingTransactionNotification = async (
+          const transactions: TransactionSummary[] = values
+            .filter((tx) => isTransactionSummary(tx))
+            .map((item: TransactionListItem) => (item as Transaction).transaction)
+          const userAddress: string = userAccountSelector(state)
+          const awaitingTransactions = getAwaitingGatewayTransactions(transactions, userAddress)
+
+          const awaitingTxsSubmissionDateList = awaitingTransactions.map((tx) => tx.timestamp)
+
+          const safesMap = safesAsMap(state)
+          const currentSafe = safesMap.get(safeAddress)
+
+          if (!currentSafe || !isUserAnOwner(currentSafe, userAddress) || awaitingTransactions.length === 0) {
+            break
+          }
+
+          const notificationKey = `${safeAddress}-awaiting`
+
+          await sendAwaitingTransactionNotification(
+            dispatch,
+            safeAddress,
+            awaitingTxsSubmissionDateList,
+            notificationKey,
+            onNotificationClicked(dispatch, notificationKey, safeAddress),
+          )
 
-        const notificationKey = `${safeAddress}-awaiting`
-
-        await sendAwaitingTransactionNotification(
-          dispatch,
-          safeAddress,
-          awaitingTxsSubmissionDateList,
-          notificationKey,
-          onNotificationClicked(dispatch, notificationKey, safeAddress),
-        )
-
-        break
-      }
-      case ADD_OR_UPDATE_SAFE: {
-        const state = store.getState()
-        const { safe } = action.payload
-        const currentSafeAddress = safeAddressFromUrl(state) || safe.address
-        if (!currentSafeAddress || !safe.currentVersion) {
           break
         }
-        const isUserOwner = grantedSelector(state)
-        const version = await getSafeVersionInfo(safe.currentVersion)
-
-        const notificationKey = `${currentSafeAddress}-update`
-        const onNotificationClicked = () => {
-          dispatch(closeSnackbarAction({ key: notificationKey }))
-          dispatch(push(`/safes/${currentSafeAddress}/settings`))
-        }
-
-        if (version?.needUpdate && isUserOwner) {
-          dispatch(
-            enqueueSnackbar(
-              enhanceSnackbarForAction(
-                NOTIFICATIONS.SAFE_NEW_VERSION_AVAILABLE,
-                notificationKey,
-                onNotificationClicked,
+        case ADD_OR_UPDATE_SAFE: {
+          const state = store.getState()
+          const { safe } = action.payload
+          const currentSafeAddress = safeAddressFromUrl() || safe.address
+          if (!currentSafeAddress || !safe.currentVersion) {
+            break
+          }
+          const isUserOwner = grantedSelector(state)
+          const version = await getSafeVersionInfo(safe.currentVersion)
+
+          const notificationKey = `${currentSafeAddress}-update`
+          const onNotificationClicked = () => {
+            dispatch(closeSnackbarAction({ key: notificationKey }))
+            history.push(`/safes/${currentSafeAddress}/settings`)
```

Comment:
Doesn't this need a network path param?

Reply:
It does. I'll add it. Thanks for spotting that.